py_ecc/optimized_bls12_381/optimized_field_elements.py

```diff
@@ -239,7 +239,10 @@ def zero(cls):
 # The quadratic extension field
 class FQ2(FQP):
     def __init__(self, coeffs):
-        self.coeffs = tuple(coeffs)
+        if isinstance(coeffs[0], int):
+            self.coeffs = tuple(coeff % field_modulus for coeff in coeffs)
```

Comment:
Wouldn't it be better to put that in the FQP class
and do:

`super(FQ2, self).__init__(coeffs, modulus_coeffs)`.

This way no need to duplicate the effort in FQ12 (and other extension fields).

Reply:
I agree with @mratsim

Reply:
Was about to do that, but realized that all __init__s are duplicated in `optimized_bls12_381`. A profile shows that using `super` in FQP would affect performance, so does the modulus check in this PR. Need to figure out a tolerable bound first. 

Master branch:
FQ2 1.123668404
FQ12 1.279635752

This PR:
FQ2 1.790521966
FQ12 2.642265235

Using Super in FQP:
FQ2 2.097532083
FQ12 3.0613951150000003

https://gist.github.com/ChihChengLiang/115ee83f1e5ce792db8ccf2392244455

Reply:
Arrrr ... my bad, no need to check type. The input is always int.

Reply:
Asked v about the performance, "get as much performance as we can" is desired.

Reply:
Asked v for review. There are other codes pass FQs as coefficients to FQ2 and FQP, so `isinstance` check is required.